lib/ansible/modules/cloud/amazon/cloudwatchlogs_log_group.py

```diff
@@ -0,0 +1,298 @@
+
+        return_create_log_group = {}
+        if 'logGroups' in desc_log_group:
+            for i in desc_log_group['logGroups']:
+                if module.params['log_group_name'] == i['logGroupName'] and module.params['overwrite'] is True:
+                    delete_log_group(client=logs, log_group_name=module.params['log_group_name'], module=module)
+                    return_create_log_group = create_log_group(client=logs,
+                                                               log_group_name=module.params['log_group_name'],
+                                                               kms_key_id=module.params['kms_key_id'],
+                                                               tags=module.params['tags'],
+                                                               retention=module.params['retention'],
+                                                               module=module)
+                elif module.params['log_group_name'] == i['logGroupName']:
+                    return_create_log_group = i
+
+        if not return_create_log_group:
+                    return_create_log_group = create_log_group(client=logs,
+                                                               log_group_name=module.params['log_group_name'],
+                                                               kms_key_id=module.params['kms_key_id'],
+                                                               tags=module.params['tags'],
+                                                               retention=module.params['retention'],
+                                                               module=module)
+
+        desc_log_group_snake = {}
+        for k, v in return_create_log_group.items():
+            desc_log_group_snake[_camel_to_snake(k)] = v
+
+        desc_log_group_result = dict(changed=False, stdout=desc_log_group_snake)
+        module.exit_json(**desc_log_group_result)
+
+    elif state == 'absent':
+        delete_log_group(client=logs,
+                         log_group_name=module.params['log_group_name'],
+                         module=module)
+    elif state == 'get':
+        desc_log_group = describe_log_group(client=logs,
+                                            log_group_name=module.params['log_group_name'],
+                                            module=module)
+
+        if 'logGroups' not in desc_log_group:
+            module.fail_json(msg="LogGroupNamePrefix does not exists.")
+
+        desc_log_group_snake = {}
+        final_log_group_snake = []
+
+        for lst in desc_log_group['logGroups']:
+            for k, v in lst.items():
+                desc_log_group_snake[_camel_to_snake(k)] = v
+            final_log_group_snake.append(desc_log_group_snake.copy())
+
+        desc_log_group_result = dict(changed=False, stdout=final_log_group_snake)
+        module.exit_json(**desc_log_group_result)
+
+    else:
```

Comment:
You can remove this else because the ec2_argument_spec enforces that one of the allowed choices is provided.